apisix/init.lua

```diff
@@ -454,8 +454,15 @@ function _M.http_access_phase()
 
             if changed then
                 api_ctx.matched_route = route
+                local runned_plugins = core.table.deepcopy(api_ctx.plugins)
                 core.table.clear(api_ctx.plugins)
-                api_ctx.plugins = plugin.filter(api_ctx, route, api_ctx.plugins)
+                local unrunn_plugins
```

Comment:
Maybe we can add a flag in https://github.com/apache/apisix/blob/c4229d15efe8cac6677f8124cf92d3512d1b0180/apisix/plugin.lua#L505 so that we can save allocation?

Reply:
update